### PR TITLE
Added helper class to handle replacing mana symbols with images

### LINKED ISSUE #4: Converting manacost to images
{W} => Water symbol
{2} => 2 image 

https://mtg.gamepedia.com/Numbers_and_symbols

### PR DESCRIPTION
**Pull request by ConnorDownes**
Created a static helper class to convert symbols to images by looping through and using a string format.

This PR fixes #4

### CODE REVIEW

File: src/MTGCore/Models/Card.cs [hunk 1]
```diff
@@ -9,7 +10,8 @@ public class Card
 
 
         public string name { get; set; }
-        public string manaCost { get; set; }
+        private string _manaCost;
+        public string manaCost { get { return _manaCost; } set { _manaCost = ManaConvert.String(value); } }
```
Comment (StephenGibbsDev) [inline src/MTGCore/Models/Card.cs]: this is a good solution however manacost is now entirely dependent on the manaconvery class. 

Maybe look into injecting dependancy into accessors? (dunno if this is possible)

Comment (StephenGibbsDev) [reply src/MTGCore/Models/Card.cs]: https://softwareengineering.stackexchange.com/questions/360525/dependency-injection-vs-static-methods

ignore me, static method id just fine. I would make a unit test tho

File: src/MTGCore/HelperClasses/ManaConvert.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            string[] sections = manaCost.Split(new string[] { "{", "}" }, StringSplitOptions.RemoveEmptyEntries);
+            string output = "";
+            foreach (string section in sections)
+            {
+                output += string.Format("<img src=\"/images/{0}.svg\" height=\"20\" width=\"20\" />", section);
+            }
+            return output;
+        }
+    }
+}
```
Comment (StephenGibbsDev) [inline src/MTGCore/HelperClasses/ManaConvert.cs]: add a unit test to my unit test project to test this.

heres the testing documentation
http://docs.shouldly-lib.net/

File: src/MTGCore/HelperClasses/ManaConvert.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public static class ManaConvert
+    {
+        public static string String (string manaCost)
+        {
+            string[] sections = manaCost.Split(new string[] { "{", "}" }, StringSplitOptions.RemoveEmptyEntries);
+            string output = "";
+            foreach (string section in sections)
+            {
+                output += string.Format("<img src=\"/images/{0}.svg\" height=\"20\" width=\"20\" />", section);
```
Comment (StephenGibbsDev) [inline src/MTGCore/HelperClasses/ManaConvert.cs]: output += string.Format($"");

above is how to do it with string interpolation

File: src/MTGCore/Models/Card.cs (hunk 1 above)
Comment (StephenGibbsDev) [reply src/MTGCore/Models/Card.cs]: I've changed my mind, making manacost dependant on the manaconvert method is spooky to me. There for i would recommend putting this method in its own service class and constructor injecting it into the controller.

you can then pass in the model and it will return a new model with a new manacost property already populated.

//this wwould be in the controller
var cardsWithSymbols = MyService.ConvertMana(cards);

Return View(cardswithSymbols)

//iinside card/cards controller wwe add new property
public string manaCostWithSymbols {get;set;}

This also means wwe keep the manacost string if we ever needed it for anything and the cards/card controller now isnt dependant on thatmethod to run.

there is plenty of documentation on DI in .net core but if you need help let me know, this would be really good practice for DI.

File: test/MTGCore.Tests/HelperClassesTest.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        [Fact]
+        public void ManaConvert()
+        {
+            string manaCost = "{W}{R}{5}";
+
+            string convertedManaCost = MTGCore.HelperClasses.ManaConvert.String(manaCost);
+
+            convertedManaCost.ShouldBe("<img src=\"/images/W.svg\" height=\"20\" width=\"20\" /><img src=\"/images/R.svg\" height=\"20\" width=\"20\" /><img src=\"/images/5.svg\" height=\"20\" width=\"20\" />");
+        }
+    }
```
Comment (StephenGibbsDev) [inline test/MTGCore.Tests/HelperClassesTest.cs]: i would add a test for iif the mana cost is something that doesnt exist eg {POOP}. or {Z}. and make sure your handling that error correctly.

Also make test testing empty string

File: src/MTGCore/Models/Card.cs (hunk 1 above)
Comment (ConnorDownes) [reply src/MTGCore/Models/Card.cs]: I'll have a look at implementing it using this approach when I have a chance.

**Comment by ConnorDownes** [comment]
I have switched the helper class for a service. I will need to modify the unit test to account for this. I think I prefer my original method but I think this way is more flexible.

File: src/MTGCore/Services/ManaConversionService.cs [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
-using System.Linq;
-using System.Threading.Tasks;
+using System.IO;
 
 namespace MTGCore.Services
 {
     public class ManaConversionService : IConversionService
     {
-        public string ConvertToSymbol(string? manaCost)
+        public string ConvertToSymbol(string? manaCost, string root)
```
[the remark this answers is not in the document]
Comment (ConnorDownes) [reply src/MTGCore/Services/ManaConversionService.cs]: Modified.

File: src/MTGCore/Controllers/CardController.cs [hunk 6]
```diff
@@ -22,21 +23,25 @@ public class CardController : Controller
         private IMapper _mapper;
         private readonly IRepoContext _context;
         private readonly IConversionService _conversion;
+        private readonly IWebHostEnvironment _env;
 
-        public CardController(MTGService mtgservice, IMapper mapper, IRepoContext context, IConversionService conversion)
+        public CardController(MTGService mtgservice, IMapper mapper, IRepoContext context, IConversionService conversion, IWebHostEnvironment env)
```
[the remark this answers is not in the document]
Comment (ConnorDownes) [reply src/MTGCore/Controllers/CardController.cs]: Alternative approach used within ManaConversion class.

File: src/MTGCore/Startup.cs [hunk 7]
```diff
@@ -42,7 +42,7 @@ public void ConfigureServices(IServiceCollection services)
             IFileService fileService = new FileService();
             services.AddScoped<IRepoContext, RepoContext>();
             //services.AddScoped<IConversionService, ManaConversionService>();
-            services.AddTransient<IConversionService>(s => new ManaConversionService(fileService));
+            services.AddTransient<IConversionService>(s => new ManaConversionService(fileService, Environment.CurrentDirectory + @"\wwwroot\images\"));
```
Comment (StephenGibbsDev) [inline src/MTGCore/Startup.cs]: change this to scoped

File: src/MTGCore/Startup.cs [hunk 8]
```diff
@@ -38,7 +39,10 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddDbContext<RepoContext>(options => options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
 
+            IFileService fileService = new FileService();
             services.AddScoped<IRepoContext, RepoContext>();
+            //services.AddScoped<IConversionService, ManaConversionService>();
+            services.AddTransient<IConversionService>(s => new ManaConversionService(fileService, Environment.CurrentDirectory + @"\wwwroot\images\"));
```
Comment (StephenGibbsDev) [inline src/MTGCore/Startup.cs]: change this to scoped
